### PR TITLE
Fix biospecimen check-in modal buttons

### PR DESCRIPTION
**Pull request by Gbarra9**
This pull request is related to the links below:
- https://github.com/episphere/connect/issues/1004
- https://github.com/episphere/biospecimen/pull/723
- https://github.com/episphere/biospecimen/pull/726

Problem
The Workflow to create the problem. A research user is successfully checked in after clicking on the check-in modal. A user then clicks either the `cancel` button or `continue` button. This does not reset or dismiss the bootstrap modal's class attributes and styling. After clicking either "cancel" or "continue", search a participant that has recorded clinical specimens collected. The check-in block does not appear when the clinical participant searched in the research dashboard workflow. The close button or "X" button properly dismisses/resets the biospecimen modal class attributes and styling.

Code Changes:
- created new function `dismissBiospecimenModal` to target biospecimen modal "close button" to dismiss/reset the modal's class attributes/ styling
- call `dismissBiospecimenModal` when either cancel or continue button is clicked
- replace areas of code with function `dismissBiospecimenModal`

Screenshot 2024-07-17 at 9 31 41 AM

### CODE REVIEW

**Review by jhflorey — APPROVED** [review]

File: src/events.js [hunk 1]
```diff
@@ -662,12 +662,15 @@ const handleCheckInModal = async (data, visitConcept, query) => {
...
     const checkInOnContinue = async () => {
         try {
             const updatedResponse = await findParticipant(query);
             const updatedData = updatedResponse.data[0];
-    
+
+            dismissBiospecimenModal();
             specimenTemplate(updatedData);
         } catch (error) {
             showNotifications({ title: 'Error', body: 'There was an error checking in the participant. Please try again.' });
```
Comment (we-ai) [inline src/events.js:676]: Instead of splitting handlers of "Cancel" and "Continue" buttons in different locations, I suggest put these click handlers inside `showNotificationsCancelOrContinue`, to make things clearer.

File: src/events.js [hunk 2]
```diff
@@ -571,10 +571,7 @@ export const addEventCheckInCompleteForm = (isCheckedIn, checkOutFlag) => {
                 hideAnimation();
                 showTimedNotifications({ title: 'Success', body: 'Participant is checked out.' }, 100000, 1500);
                 setTimeout(() => {
-                    const closeButton = document.querySelector('#biospecimenModal .btn[data-dismiss="modal"]');
-                    if (closeButton) {
-                        closeButton.click();
-                    }
+                    dismissBiospecimenModal();
```
Comment (we-ai) [inline src/events.js:574]: Why is this line needed? `showTimedNotifications` calls `dismissBiospecimenModal` and runs before this line.

File: src/events.js [hunk 3]
```diff
@@ -623,10 +621,12 @@ const checkClinicalBloodOrUrineCollected = (participantData) => {
         collectionDetailsBaseline?.[conceptIds.clinicalSiteUrineRRLReceived]
     ];
 
-    if (collectedBaselineStatuses.includes(conceptIds.yes)) { 
+    if (collectedBaselineStatuses.includes(conceptIds.yes)) {
         const modalIcon = `<i class="fas fa-exclamation-circle" style="color: red; font-size: 1.4rem;"></i>`
         const bodyMessage = `Check In not allowed, participant already has clinical collection for this timepoint. If you have questions, contact the Connect Biospeicmen Team: <a href="mailto:connectbioteam@nih.gov">connectbioteam@nih.gov</a>.`
+
         showNotifications({ title: `${modalIcon} WARNING`, body: bodyMessage });
+        const biospecimenModal = document.getElementById('biospecimenModal');
```
Comment (we-ai) [inline src/events.js:629]: What's the use of this line?

File: src/events.js (hunk 1 above)
Comment (Gbarra9) [reply src/events.js:676]: This function is used in multiple places. This will need a refactor in the future.

**Review by we-ai — APPROVED** [review]
Looks good.
